docs/src/index.md

```diff
@@ -193,7 +193,292 @@ Since we haven't created our own project yet, we are in the default project, loc
 To return to the `julia>` prompt, either press backspace when the input line is empty or press Ctrl+C.
 Help is available by calling `pkg> help`.
 
-To generate files for a new project, use `pkg> generate`.
+
+### Adding packages
+
+There are two ways of adding packages, either using the `add` command or the `dev` command.
+The most frequently used one is `add` and its usage is described first.
+
+#### Adding registered packages
+
+```
+(v0.7) pkg> add Example
+   Cloning default registries into /Users/kristoffer/.julia/registries
+   Cloning registry Uncurated from "https://github.com/JuliaRegistries/Uncurated.git"
+  Updating registry at `~/.julia/registries/Uncurated`
+  Updating git-repo `https://github.com/JuliaRegistries/Uncurated.git`
+ Resolving package versions...
+  Updating `~/.julia/environments/v0.7/Project.toml`
+  [7876af07] + Example v0.5.1
+  Updating `~/.julia/environments/v0.7/Manifest.toml`
+  [7876af07] + Example v0.5.1
+  [8dfed614] + Test
+```
+
```

Comment:
In the very short time I've spent with the new `Pkg`, one of the biggest comprehension hurdles has been trying to understand/gain intuition for the differences/relationships between "projects", "environments", "registries" and "packages". For this reason, it might be helpful to have some explicit clarifying language immediately following this code example, e.g. 

"Here, we add the package `Example` to the current *project*.  In this example, we are using a fresh Julia installation, and this is our first time `add`ing a package using `Pkg`. By default, `Pkg` clones Julia's `Uncurated` *registry*, and uses this registry to look up packages requested for inclusion in the current *environment*." 

For all I know, this specific explanation/language isn't even correct, but you get my point 😛

Reply:
There is a glossary for this very reason: https://docs.julialang.org/en/latest/stdlib/Pkg/#Glossary-1

Reply:
Right, I've read the glossary, but I'm not confident that I know how to use these words/concepts correctly just given their concise technical definitions. I'm sure many other users will feel the same.

This section (and the other sections here, since this seems to be a proper manual rather than straight API documentation) is an excellent place to demonstrate these words/concepts to use in context.